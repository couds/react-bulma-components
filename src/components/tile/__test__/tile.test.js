import React from 'react';
import renderer from 'react-test-renderer';
import Tile from '..';
import CONSTANTS from '../../../constants';

describe('Tile component', () => {
  it('Should exist', () => {
    expect(Tile).toMatchSnapshot();
  });
  it('Should have notification classname', () => {
    const component = renderer.create(
      <Tile>
        <img
          alt="placeholder"
          src="http://bulma.io/images/placeholders/128x128.png"
        />
      </Tile>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat classname in props with Bulma classname', () => {
    const component = renderer.create(
      <Tile className="other-class this-is-a-test">
        <p>Default</p>
      </Tile>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should use inline styles', () => {
    const component = renderer.create(
      <Tile style={{ height: 250 }}>
        <p>Default</p>
      </Tile>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should render as Section', () => {
    const component = renderer.create(
      <Tile renderAs="section">
        <p>Default</p>
      </Tile>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  [null]
    .concat(Object.keys(CONSTANTS.COLORS).map(key => CONSTANTS.COLORS[key]))
    .map(color =>
      it(`Should use use color ${color}`, () => {
        const component = renderer.create(
          <Tile notification color={color}>
            <p>Default</p>
          </Tile>,
        );
        expect(component.toJSON()).toMatchSnapshot();
      }),
    );
});
