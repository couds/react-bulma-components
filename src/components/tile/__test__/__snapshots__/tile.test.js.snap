// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile component Should concat classname in props with Bulma classname 1`] = `
<div
  className="tile other-class this-is-a-test"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Tile component Should exist 1`] = `[Function]`;

exports[`Tile component Should have notification classname 1`] = `
<div
  className="tile"
  style={Object {}}
>
  <img
    alt="placeholder"
    src="http://bulma.io/images/placeholders/128x128.png"
  />
</div>
`;

exports[`Tile component Should render as Section 1`] = `
<section
  className="tile"
  style={Object {}}
>
  <p>
    Default
  </p>
</section>
`;

exports[`Tile component Should use inline styles 1`] = `
<div
  className="tile"
  style={
    Object {
      "height": 250,
    }
  }
>
  <p>
    Default
  </p>
</div>
`;

exports[`Tile component Should use use color black 1`] = `
<div
  className="tile notification is-black"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Tile component Should use use color danger 1`] = `
<div
  className="tile notification is-danger"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Tile component Should use use color dark 1`] = `
<div
  className="tile notification is-dark"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Tile component Should use use color info 1`] = `
<div
  className="tile notification is-info"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Tile component Should use use color light 1`] = `
<div
  className="tile notification is-light"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Tile component Should use use color null 1`] = `
<div
  className="tile notification"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Tile component Should use use color primary 1`] = `
<div
  className="tile notification is-primary"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Tile component Should use use color success 1`] = `
<div
  className="tile notification is-success"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Tile component Should use use color warning 1`] = `
<div
  className="tile notification is-warning"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;

exports[`Tile component Should use use color white 1`] = `
<div
  className="tile notification is-white"
  style={Object {}}
>
  <p>
    Default
  </p>
</div>
`;
